surveyor/collector_statz.go

```diff
@@ -868,198 +867,250 @@ func (sc *StatzCollector) Describe(ch chan<- *prometheus.Desc) {
-			}
-			for tier, size := range stat.jetstreamTieredStorageReserved {
-				ch <- newGaugeMetric(sc.descs.accJetstreamTieredStorageReserved, size, append(id, fmt.Sprintf("R%d", tier)))
+		// Account scope metrics
+		if sc.collectAccounts {
+			metrics.newGaugeMetric(sc.descs.accCount, float64(len(sc.accStats)), nil)
+			for _, stat := range sc.accStats {
+				id := []string{stat.accountID}
+
+				metrics.newGaugeMetric(sc.descs.accConnCount, stat.connCount, id)
+				metrics.newGaugeMetric(sc.descs.accLeafCount, stat.leafCount, id)
+				metrics.newGaugeMetric(sc.descs.accSubCount, stat.subCount, id)
+
+				metrics.newCounterMetric(sc.descs.accBytesSent, stat.bytesSent, id)
+				metrics.newCounterMetric(sc.descs.accBytesRecv, stat.bytesRecv, id)
+				metrics.newCounterMetric(sc.descs.accMsgsSent, stat.msgsSent, id)
+				metrics.newCounterMetric(sc.descs.accMsgsRecv, stat.msgsRecv, id)
+
+				metrics.newGaugeMetric(sc.descs.accJetstreamEnabled, stat.jetstreamEnabled, id)
+				metrics.newGaugeMetric(sc.descs.accJetstreamMemoryUsed, stat.jetstreamMemoryUsed, id)
+				metrics.newGaugeMetric(sc.descs.accJetstreamStorageUsed, stat.jetstreamStorageUsed, id)
+				metrics.newGaugeMetric(sc.descs.accJetstreamMemoryReserved, stat.jetstreamMemoryReserved, id)
+				metrics.newGaugeMetric(sc.descs.accJetstreamStorageReserved, stat.jetstreamStorageReserved, id)
+				for tier, size := range stat.jetstreamTieredMemoryUsed {
+					metrics.newGaugeMetric(sc.descs.accJetstreamTieredMemoryUsed, size, append(id, fmt.Sprintf("R%d", tier)))
+				}
+				for tier, size := range stat.jetstreamTieredStorageUsed {
+					metrics.newGaugeMetric(sc.descs.accJetstreamTieredStorageUsed, size, append(id, fmt.Sprintf("R%d", tier)))
+				}
+				for tier, size := range stat.jetstreamTieredMemoryReserved {
+					metrics.newGaugeMetric(sc.descs.accJetstreamTieredMemoryReserved, size, append(id, fmt.Sprintf("R%d", tier)))
+				}
+				for tier, size := range stat.jetstreamTieredStorageReserved {
+					metrics.newGaugeMetric(sc.descs.accJetstreamTieredStorageReserved, size, append(id, fmt.Sprintf("R%d", tier)))
+				}
+
+				metrics.newGaugeMetric(sc.descs.accJetstreamStreamCount, stat.jetstreamStreamCount, id)
+				for _, streamStat := range stat.jetstreamStreams {
+					metrics.newGaugeMetric(sc.descs.accJetstreamConsumerCount, streamStat.consumerCount, append(id, streamStat.streamName))
+					metrics.newGaugeMetric(sc.descs.accJetstreamReplicaCount, streamStat.replicaCount, append(id, streamStat.streamName))
+				}
 			}
+		}
 
-			ch <- newGaugeMetric(sc.descs.accJetstreamStreamCount, stat.jetstreamStreamCount, id)
-			for _, streamStat := range stat.jetstreamStreams {
-				ch <- newGaugeMetric(sc.descs.accJetstreamConsumerCount, streamStat.consumerCount, append(id, streamStat.streamName))
-				ch <- newGaugeMetric(sc.descs.accJetstreamReplicaCount, streamStat.replicaCount, append(id, streamStat.streamName))
+		bufferChan := make(chan prometheus.Metric)
```

Comment:
`bufferChan` seems like a bad name here because this is an un-buffered channel...  maybe call it `collectCh`?

Reply:
That's fair - I only named it that way since it's basically a buffer before writing out to the real channel.